de.fu_berlin.inf.dpp.whiteboard/src/de/fu_berlin/inf/dpp/whiteboard/ui/browser/BrowserActionExecuter.java

```diff
@@ -0,0 +1,103 @@
+import de.fu_berlin.inf.dpp.whiteboard.ui.browser.BrowserAction.ActionTypes;
+
+public class BrowserActionExecuter {
+
+    private static final Logger LOG = Logger
+        .getLogger(BrowserActionExecuter.class);
+
+    private final SXEController sxe;
+
+    public BrowserActionExecuter(SXEController sxe) {
+        this.sxe = sxe;
+    }
+
+    public void execute(BrowserAction action) {
+        sxe.executeAndCommit(getChangeRecords(action));
+    }
+
+    private List<IRecord> getChangeRecords(BrowserAction action) {
+        String type = action.getType();
+        if (type.equals(ActionTypes.NEW)) {
+            return create(action);
+        } else if (type.equals(ActionTypes.SET)) {
+            return set(action);
+        } else if (type.equals(ActionTypes.REMOVE)) {
+            return remove(action);
+        } else {
+            LOG.error("undefined action type received: " + type);
+            return new LinkedList<IRecord>();
+        }
+    }
+
+    private List<IRecord> create(BrowserAction action) {
+        List<IRecord> records = new LinkedList<IRecord>();
+        // since the library fabric.js (used in the html whiteboard) does not
+        // support parent hierarchies this function will always add the new
+        // element to the root element
+        ElementRecord parent = sxe.getDocumentRecord().getRoot();
+        ElementRecord newElement = parent.createNewElementRecord(null,
+            action.getElementType());
+        newElement.setRid(action.getID());
+        records.add(newElement);
+        records.addAll(setAttributes(newElement, action.getAttributes()));
+        return records;
+    }
+
+    private List<IRecord> setAttributes(ElementRecord element,
+        ArrayList<Map<String, String>> attributes) {
+        List<IRecord> records = new LinkedList<IRecord>();
+        for (Map<String, String> attribute : attributes) {
+            IRecord record = element.createNewOrSetAttributeRecord(null,
+                attribute.get("name"), attribute.get("chdata"), false);
+            if (record instanceof AttributeRecord)
+                ((AttributeRecord) record).setRid(attribute.get("rid"));
+            records.add(record);
+        }
+        return records;
+    }
+
+    private List<IRecord> set(BrowserAction action) {
+        ElementRecord element = getElement(action.getID());
+        if (element == null)
+            return new LinkedList<IRecord>();
+        return setAttributes(element, action.getAttributes());
+
+    }
+
+    private List<IRecord> remove(BrowserAction action) {
+        List<IRecord> records = new LinkedList<IRecord>();
+        ElementRecord element = getElement(action.getID());
+        if (element == null)
+            return new LinkedList<IRecord>();
+
+        records.add(element.getRemoveRecord());
+        return records;
+    }
+
+    private ElementRecord getElement(String id) {
+        try {
+            ElementRecord element = sxe.getDocumentRecord()
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Immediately return this expression instead of assigning it to the temporary variable "element". ([squid:S1488](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1488))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=250&issues=AWY_InU-nzfF1D4JrUtb&open=AWY_InU-nzfF1D4JrUtb)

[ISSUE_KEY]: AWY_InU-nzfF1D4JrUtb